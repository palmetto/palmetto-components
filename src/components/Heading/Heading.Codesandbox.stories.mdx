import { Meta } from '@storybook/addon-docs/blocks';

<Meta title="Components/Heading/Codesandbox" />

# Heading

Use `Headings` as labels for pages or sections of a page that make up an interface. `Headings` can label an entire page, or sections of related content.
The `Heading` component allows you to specify an appropriate semantic element (h1-h6) and independently set its size so that it is appropriate for the surrounding content.
If no size is specified, a default size will be applied.

```js codesandbox=palmetto-components
import React from 'react';
import { Heading } from '@palmetto/palmetto-components';

export default () => <Heading as="h1">Lead the world towards a clean energy future</Heading>;
```

## All Levels

Each Heading level has a default size.

```js codesandbox=palmetto-components
import React from 'react';
import { Heading } from '@palmetto/palmetto-components';

export default () => (
  <>
    <Heading as="h1">h1 Heading</Heading>
    <Heading as="h2">h2 Heading</Heading>
    <Heading as="h3">h3 Heading</Heading>
    <Heading as="h4">h4 Heading</Heading>
    <Heading as="h5">h5 Heading</Heading>
    <Heading as="h6">h6 Heading</Heading>
  </>
);
```

## Sizes

The default size of the Heading can overridden as xs-5xl so that it's appropriate for its surrounding context.

```js codesandbox=palmetto-components
import React from 'react';
import { Heading, FONT_SIZE_OPTIONS } from '@palmetto/palmetto-components';

export default () => (
  <>
    {[...FONT_SIZE_OPTIONS].reverse().map(fontSize => (
      <Heading as="h3" size={fontSize} key={fontSize}>
        {`Size ${fontSize}`}
      </Heading>
    ))}
  </>
);
```

The `size` prop accepts an object where the properties correspond to [breakpoint tokens](?path=/story/design-tokens-design-tokens--page#breakpoints).

```js codesandbox=palmetto-components
import React from 'react';
import { Heading } from '@palmetto/palmetto-components';

export default () => (
  <Heading as="h3" size={{ base: 'sm', tablet: 'lg', desktop: 'xl', hd: '3xl' }}>
    Leading the world towards a clean energy future
  </Heading>
);
```

## Colors

The variant of the heading is inherited from its parent, but can be set to brand variant as well.

```js codesandbox=palmetto-components
import React from 'react';
import { Heading, FONT_COLOR_OPTIONS } from '@palmetto/palmetto-components';

export default () => (
  <>
    {FONT_COLOR_OPTIONS.map(variant => (
      <Heading as="h3" variant={variant} key={variant}>
        {variant}
      </Heading>
    ))}
  </>
);
```
