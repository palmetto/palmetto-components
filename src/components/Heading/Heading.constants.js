export const HEADING_LEVELS = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6']; // eslint-disable-line import/prefer-default-export
