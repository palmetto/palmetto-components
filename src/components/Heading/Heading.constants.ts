export type HEADING_LEVELS_TYPE = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';

export const HEADING_LEVELS: HEADING_LEVELS_TYPE[] = [
  'h1',
  'h2',
  'h3',
  'h4',
  'h5',
  'h6',
]; // eslint-disable-line import/prefer-default-export

export const HEADING_DEFAULT_SIZE_MAP = {
  h1: '3xl',
  h2: '2xl',
  h3: 'xl',
  h4: 'lg',
  h5: 'md',
  h6: 'sm',
};
