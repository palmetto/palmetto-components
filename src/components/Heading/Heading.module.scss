.heading {
  line-height: var(--size-line-height-heading);
  font-family: Nexa, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu,
    Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: var(--size-font-weight-black);
}
