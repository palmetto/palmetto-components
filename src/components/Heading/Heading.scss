.Palmetto-Heading {
  line-height: 1; // @TODO -- figure out specific line-heights for each heading.
}
