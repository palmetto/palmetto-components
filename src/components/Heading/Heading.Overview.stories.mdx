import { Heading } from './Heading';
import { v4 as uuid } from 'uuid';
import { action } from '@storybook/addon-actions';
import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs';
import { FONT_SIZE_OPTIONS, FONT_COLOR_OPTIONS } from '../../lib/tokens';
import { RESPONSIVE_STORY } from '../../docs/constants';

<Meta title="Components/Heading/Overview" component={Heading} />

# Heading

Use `Headings` as labels for pages or sections of a page that make up an interface. `Headings` can label an entire page, or sections of related content.
The `Heading` component allows you to specify an appropriate semantic element (h1-h6) and independently set its size so that it is appropriate for the surrounding content.
If no size is specified, a default size will be applied.

<Canvas isExpanded>
  <Heading as="h1">Lead the world towards a clean energy future</Heading>
</Canvas>

## Props

<ArgsTable of={Heading} />

## All Levels

Each Heading level has a default size.

<Canvas>
  <Story
    name="Levels"
    parameters={{
      design: {
        type: 'figma',
        url: 'https://www.figma.com/file/k1kOsplTGXvSSMQQCzhCND/Palmetto-Design-System?node-id=742%3A8800',
      },
    }}
  >
    <>
      <Heading as="h1">h1 Heading</Heading>
      <Heading as="h2">h2 Heading</Heading>
      <Heading as="h3">h3 Heading</Heading>
      <Heading as="h4">h4 Heading</Heading>
      <Heading as="h5">h5 Heading</Heading>
      <Heading as="h6">h6 Heading</Heading>
    </>
  </Story>
</Canvas>

## Sizes

The default size of the Heading can overridden as xs-5xl so that it's appropriate for its surrounding context.

<Canvas>
  <Story name="Sizes">
    <>
      {[...FONT_SIZE_OPTIONS].reverse().map(fontSize => (
        <Heading as="h3" size={fontSize} key={uuid()}>
          {`Size ${fontSize}`}
        </Heading>
      ))}
    </>
  </Story>
</Canvas>

The `size` prop accepts an object where the properties correspond to [breakpoint tokens](?path=/story/design-tokens-design-tokens--page#breakpoints).

<Canvas>
  <Story name="Responsive Sizes" parameters={{ ...RESPONSIVE_STORY }}>
    <Heading
      as="h3"
      size={{ base: 'sm', tablet: 'lg', desktop: 'xl', hd: '3xl' }}
    >
      Leading the world towards a clean energy future
    </Heading>
  </Story>
</Canvas>

## Colors

The variant of the heading is inherited from its parent, but can be set to brand variant as well.

<Canvas>
  <Story name="Variant">
    <>
      {FONT_COLOR_OPTIONS.map(variant => (
        <Heading as="h3" variant={variant} key={uuid()}>
          {variant}
        </Heading>
      ))}
    </>
  </Story>
</Canvas>

## Component Design Tokens

<table>
  <thead>
    <tr>
      <th>token name</th>
      <th>default value</th>
    </tr>
  </thead>
  <tbody>
    {(() => {
      const tokens = {
      '--heading-font-family': '--asset-fonts-brand',
      '--heading-font-weight': '--size-font-weight-black',
      '--heading-line-height': '--size-line-height-heading',
      };
      return Object.entries(tokens).map(([name, entry], i) => (
        <tr key={i}>
          <td>
            <code>{name}</code>
          </td>
          <td>
            <code>{entry}</code>
          </td>
        </tr>
      ));
    })()}

  </tbody>
</table>
