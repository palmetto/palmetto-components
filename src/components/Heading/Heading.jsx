import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { PALMETTO_FONT_SIZE_OPTIONS, PALMETTO_BRAND_COLOR_OPTIONS } from '../../lib/tokens';
import getElementType from '../../lib/getElementType';
import './Heading.scss';
import { HEADING_LEVELS } from './Heading.constants';

/**
 * Use `Headings` as labels for pages or sections of a page that make up an interface.
 * `Headings` can label an entire page, or sections of related content.
 *
 * The `Heading` component allows you to specify an appropriate semantic element (h1-h6)
 * and independently set its size so that it is appropriate for the surrounding content.
 * If no size is specified, a default size will be applied.
 */
const Heading = ({
  as,
  className,
  children,
  color,
  size,
}) => {
  const Element = getElementType(Heading, { as });

  const classes = classNames('Palmetto-Heading', className, 'heading', {
    [`font-size-${size}`]: size,
    [`font-color-${color}`]: color,
  });

  return <Element className={classes}>{children}</Element>;
};

Heading.propTypes = {
  /**
   * The DOM tag or react component to use for the element.
   * Select the appropriate semantic element (h1-h6).
   */
  as: PropTypes.oneOf(HEADING_LEVELS), // eslint-disable-line react/no-unused-prop-types
  /**
   * Additional class names to add
   */
  className: PropTypes.string,
  /**
   * Heading contents
   */
  children: PropTypes.node.isRequired,
  /**
   * A color token identifier to use for the text color.
   */
  color: PropTypes.oneOf(PALMETTO_BRAND_COLOR_OPTIONS),
  /**
   * By default, size is determined by the chosen tag (e.g. h1 is bigger than h2).
   * However, size can be set independently so that its size is appropriate for the surrounding content.
   */
  size: PropTypes.oneOf(PALMETTO_FONT_SIZE_OPTIONS),
};

Heading.defaultProps = {
  as: 'h4',
  className: '',
  color: 'dark',
  size: 'lg',
};

export default Heading;
