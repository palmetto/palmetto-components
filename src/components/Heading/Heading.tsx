import { FC, createElement, ReactNode } from 'react';
import classNames from 'classnames';
import getElementType from '../../lib/getElementType';
import { FontColor, FontSize, ResponsiveProp } from '../../types';
import { HEADING_LEVELS_TYPE, HEADING_DEFAULT_SIZE_MAP } from './Heading.constants';
import generateResponsiveClasses from '../../lib/generateResponsiveClasses';
import styles from './Heading.module.scss';

export interface HeadingProps {
  /**
   * The DOM tag or react component to use for the element.
   * Select the appropriate semantic element (h1-h6).
   */
  as?: HEADING_LEVELS_TYPE;
  /**
   * Content of the heading. Can be a string or any valid DOM node.
   */
  children?: ReactNode;
  /**
   * Additional class names to add.
   */
  className?: string;
  /**
   * A variant token identifier to use for the text variant. Available variants found:
   * [here](https://github.com/palmetto/palmetto-design-tokens/blob/main/properties/color/font.json).
   */
  variant?: FontColor;
  /**
   * By default, size is determined by the chosen tag (e.g. h1 is bigger than h2).
   * However, size can be set independently so that its size is appropriate for the surrounding content.
   * Available sizes found:
   * [here](https://github.com/palmetto/palmetto-design-tokens/blob/main/properties/size/font.json).
   */
  size?: FontSize | ResponsiveProp<FontSize>;
  /**
   * Additional props to be spread to rendered element
   */
  [x: string]: any; // eslint-disable-line
}

export const Heading: FC<HeadingProps> = ({
  as = 'h4',
  children,
  className,
  variant,
  size,
  ...restProps
}) => {
  const element = getElementType(Heading, { as });

  const headingSize = size || HEADING_DEFAULT_SIZE_MAP[as];

  const classes = classNames(
    styles.heading,
    className,
    generateResponsiveClasses('font-size', headingSize),
    {
      [`font-color-${variant}`]: variant,
    },
  );

  return createElement(element, { className: classes, children, ...restProps });
};
