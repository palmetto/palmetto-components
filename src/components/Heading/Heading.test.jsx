import React from 'react';
import { v4 as uuid } from 'uuid';
import { render } from '@testing-library/react';
import { PALMETTO_FONT_SIZE_OPTIONS, PALMETTO_BRAND_COLOR_OPTIONS } from '../../lib/tokens';
import { HEADING_LEVELS } from './Heading.constants';
import Heading from './Heading';

const headingComponentLevels = (
  HEADING_LEVELS.map(level => (
    <Heading as={level} key={uuid()}>
      {level}
      &nbsp;Heading Default Size
    </Heading>
  ))
);

const headingComponentSizes = (
  PALMETTO_FONT_SIZE_OPTIONS.map(fontSize => (
    <Heading size={fontSize} key={uuid()}>
      Size&nbsp;
      {fontSize}
    </Heading>
  ))
);

const headingComponentSizeOverrides = (
  HEADING_LEVELS.map((level, index) => (
    <Heading as={level} size={PALMETTO_FONT_SIZE_OPTIONS[index]} key={uuid()}>
      {level}
      &nbsp;Heading with&nbsp;
      {PALMETTO_FONT_SIZE_OPTIONS[index]}
      &nbsp;size
    </Heading>
  ))
);

const headingComponentColors = (
  PALMETTO_BRAND_COLOR_OPTIONS.map(color => (
    <Heading color={color} key={uuid()}>
      {color}
      &nbsp;heading
    </Heading>
  ))
);

describe('Heading', () => {
  test('Matches the snapshot for component with default props', () => {
    const { asFragment } = render(<Heading>Hello World</Heading>);
    expect(asFragment(<Heading>Hello World</Heading>)).toMatchSnapshot();
  });
  headingComponentLevels.forEach((headingComponent, index) => {
    test(`Matches the snapshot for heading ${HEADING_LEVELS[index]}`, () => {
      const { asFragment } = render(headingComponent);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  headingComponentSizes.forEach((headingComponent, index) => {
    test(`Matches the snapshot for size ${PALMETTO_FONT_SIZE_OPTIONS[index]}`, () => {
      const { asFragment } = render(headingComponent);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  headingComponentSizeOverrides.forEach((headingComponent, index) => {
    test(`Matches the snapshot for heading ${HEADING_LEVELS[index]} and font size ${PALMETTO_FONT_SIZE_OPTIONS[index]}`, () => { // eslint-disable-line max-len
      const { asFragment } = render(headingComponent);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  headingComponentColors.forEach((headingComponent, index) => {
    test(`Matches the snapshot for color ${PALMETTO_BRAND_COLOR_OPTIONS[index]}`, () => {
      const { asFragment } = render(headingComponent);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
