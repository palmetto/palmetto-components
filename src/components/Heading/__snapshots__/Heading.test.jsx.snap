// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Matches the snapshot for color black 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-lg font-color-black"
  >
    black heading
  </h4>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for color danger 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-lg font-color-danger"
  >
    danger heading
  </h4>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for color dark 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-lg font-color-dark"
  >
    dark heading
  </h4>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for color grey 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-lg font-color-grey"
  >
    grey heading
  </h4>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for color info 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-lg font-color-info"
  >
    info heading
  </h4>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for color light 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-lg font-color-light"
  >
    light heading
  </h4>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for color primary 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-lg font-color-primary"
  >
    primary heading
  </h4>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for color secondary 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-lg font-color-secondary"
  >
    secondary heading
  </h4>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for color success 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-lg font-color-success"
  >
    success heading
  </h4>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for color tertiary 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-lg font-color-tertiary"
  >
    tertiary heading
  </h4>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for color warning 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-lg font-color-warning"
  >
    warning heading
  </h4>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for component with default props 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-lg font-color-dark"
  >
    Hello World
  </h4>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for heading h1 1`] = `
<DocumentFragment>
  <h1
    class="Palmetto-Heading heading font-size-lg font-color-dark"
  >
    h1 Heading Default Size
  </h1>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for heading h1 and font size xs 1`] = `
<DocumentFragment>
  <h1
    class="Palmetto-Heading heading font-size-xs font-color-dark"
  >
    h1 Heading with xs size
  </h1>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for heading h2 1`] = `
<DocumentFragment>
  <h2
    class="Palmetto-Heading heading font-size-lg font-color-dark"
  >
    h2 Heading Default Size
  </h2>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for heading h2 and font size sm 1`] = `
<DocumentFragment>
  <h2
    class="Palmetto-Heading heading font-size-sm font-color-dark"
  >
    h2 Heading with sm size
  </h2>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for heading h3 1`] = `
<DocumentFragment>
  <h3
    class="Palmetto-Heading heading font-size-lg font-color-dark"
  >
    h3 Heading Default Size
  </h3>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for heading h3 and font size md 1`] = `
<DocumentFragment>
  <h3
    class="Palmetto-Heading heading font-size-md font-color-dark"
  >
    h3 Heading with md size
  </h3>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for heading h4 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-lg font-color-dark"
  >
    h4 Heading Default Size
  </h4>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for heading h4 and font size lg 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-lg font-color-dark"
  >
    h4 Heading with lg size
  </h4>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for heading h5 1`] = `
<DocumentFragment>
  <h5
    class="Palmetto-Heading heading font-size-lg font-color-dark"
  >
    h5 Heading Default Size
  </h5>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for heading h5 and font size xl 1`] = `
<DocumentFragment>
  <h5
    class="Palmetto-Heading heading font-size-xl font-color-dark"
  >
    h5 Heading with xl size
  </h5>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for heading h6 1`] = `
<DocumentFragment>
  <h6
    class="Palmetto-Heading heading font-size-lg font-color-dark"
  >
    h6 Heading Default Size
  </h6>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for heading h6 and font size 2xl 1`] = `
<DocumentFragment>
  <h6
    class="Palmetto-Heading heading font-size-2xl font-color-dark"
  >
    h6 Heading with 2xl size
  </h6>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for size 2xl 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-2xl font-color-dark"
  >
    Size 2xl
  </h4>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for size 3xl 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-3xl font-color-dark"
  >
    Size 3xl
  </h4>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for size 4xl 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-4xl font-color-dark"
  >
    Size 4xl
  </h4>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for size 5xl 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-5xl font-color-dark"
  >
    Size 5xl
  </h4>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for size base 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-base font-color-dark"
  >
    Size base
  </h4>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for size lg 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-lg font-color-dark"
  >
    Size lg
  </h4>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for size md 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-md font-color-dark"
  >
    Size md
  </h4>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for size sm 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-sm font-color-dark"
  >
    Size sm
  </h4>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for size xl 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-xl font-color-dark"
  >
    Size xl
  </h4>
</DocumentFragment>
`;

exports[`Heading Matches the snapshot for size xs 1`] = `
<DocumentFragment>
  <h4
    class="Palmetto-Heading heading font-size-xs font-color-dark"
  >
    Size xs
  </h4>
</DocumentFragment>
`;
